pkg/daemon/update.go

```diff
@@ -186,6 +186,7 @@ func (dn *Daemon) compareMachineConfig(oldConfig, newConfig *mcfgv1.MachineConfi
 		klog.Infof("No changes from %s to %s", oldConfigName, newConfigName)
 		return false, nil
 	}
+	klog.Infof("Changes detected from %s to %s: %+v", oldConfigName, newConfigName, mcDiff)
```

Comment:
For debugging purpose, let's call logSystem() here for logging so that this message gets captured in journal log.